### ISSUE TITLE
waiting for an advancement choice

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#3483]
I saw something strange in a game with five sides, where sides 1 and 5 where AIs.
It was shown that we are waiting for an advancement decision of side 1 – unfortunately I didn't see which unit it was.
Shortly afterwards there was this message shown for side 4 – meaning that it is not off by one.

It is unlikely that side 1 did advance a unit…
Maybe there is a bug.

**Comment by gfgtdf** [comment]
Technically an advancement choice is done by the currently playing side not the side that owns the unit so maybe a defending unit advanced.

**Comment by sevu** [comment]
I saw it again, and you are right, it was a defending unit advancing. It had just one possible advancement though…

**Comment by gfgtdf** [comment]
> It had just one possible advancement though…

yes it still uses the syncecd user choice mechanism, we also cannot change that in 1.14 since it would casue OOS with older clients then that still expect the choice. We might change that for 1.15 though.

**Comment by Wedge009** [comment]
So waiting for 1.16.0 to update this?

**Comment by gfgtdf** [comment]
Hmm no no need to wait, anyone feel free to open a PR against 1.15 for.this one I'd say.
